### PR TITLE
feat: Add type Json in variations

### PR DESCRIPTION
**Pull request by killbond007**
## Description

Capture d’écran 2021-04-03 à 23 50 44

## Links

- https://github.com/stargately/launch-room/issues/80
-

### CODE REVIEW

File: src/api-gateway/resolvers/flag-resolver.ts [hunk 1]
```diff
@@ -157,8 +158,8 @@ class FlagDetails {
   @Field(() => Int)
   offVariation: number;
 
-  @Field(() => [Boolean])
-  variations: Array<boolean>;
+  @Field(() => [GraphQLJSON || Boolean])
+  variations: any;
```
Comment (puncsky) [inline src/api-gateway/resolvers/flag-resolver.ts:162]: Array<boolean | Record<string, unknown>>

File: src/shared/feature-flags/new-flag-form.tsx [hunk 2]
```diff
@@ -42,31 +44,66 @@ export const NewFlagForm: React.FC<Props> = ({
...
+
+      case "Json":
+        return (
+          <Form.Item
+            name={["variationsJson", i]}
+            validateFirst
+            rules={[
+              { required: true, message: t("feature_flags.rule_required") },
+              () => ({
+                validator(_: any, value: string) {
```
Comment (puncsky) [inline src/shared/feature-flags/new-flag-form.tsx:69]: prefer using async keyword to promise

**Comment by bitQ2019** [comment]
image
Shows [Object Object] here, is that right?

File: src/model/flag-model.ts [hunk 3]
```diff
@@ -73,7 +73,7 @@ export class Flag extends TimeStamps {
   offVariation?: number;
 
   @prop()
-  variations: Array<boolean>;
+  variations: Array<any | any>;
```
Comment (bitQ2019) [inline src/model/flag-model.ts:76]: boolean | string ?

**Review by bitQ2019 — APPROVED** [review]

**Comment by killbond007** [comment]
> image
> Shows [Object Object] here, is that right?

I think this is problem field fallthrough.variation and field offVariation , they should support type variation. like launchdarkly
Capture d’écran 2021-04-06 à 17 28 14
Capture d’écran 2021-04-06 à 17 28 05

and in table should display field offVariation 
Capture d’écran 2021-04-06 à 17 29 06

i think we can open some issue for it  ? @puncsky

**Review by puncsky — APPROVED** [review]

**Pull request merged**